api/weather/implementations.go

```diff
@@ -17,9 +17,10 @@ type Args struct {
 
 // Get hits the openweathermap.org API to get weather data
 func (o *OpenWeatherResponse) Get(a interface{}) {
-	args, ok := a.(Args)
-	apikey := args.Apikey
-	zipcode := args.Zipcode
+	args, ok := a.(API)
+	fmt.Println("Getting weather with args: ", args)
```

Comment:
should be using a logger IMO, instead of just printing

Reply:
I will change all those. I can also try to clean up some of the messiness while I'm at it

Reply:
I've replaced all `fmt.Print` with the `log` equivalents. It's still a little messy, and it's not 100% what we want, but I think it's stable enough to merge